artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/context/AbstractConnectionContext.java

```diff
@@ -73,6 +76,8 @@ public AbstractConnectionContext(AMQPConnectionCallback connectionCallback,
                                     ScheduledExecutorService scheduledPool) {
       this.connectionCallback = connectionCallback;
       this.containerId = (containerId != null) ? containerId : UUID.randomUUID().toString();
+      connectionProperties.put(Symbol.valueOf("product"), "apache-activemq-artemis");
```

Comment:
The version string is overrideable I believe so if you had some custom patched version you could indicate that. The brokers in Qpid don't make this value customisable, so I didn't personally think it was needed. I have no objection though if others have a use case for altering that.